libft/sources/strings/strings_internal.h

```diff
@@ -0,0 +1,38 @@
+#ifndef STRINGS_INTERNAL_H
+#define STRINGS_INTERNAL_H
+
+# include <stddef.h>
+# include <stdlib.h>
+# include "strings.h"
+# include "../memory/memory.h"
+
+typedef struct s_delim
+{
+	char *delim;
+} t_delim;
+
+typedef struct s_delims
+{
+	t_delim		*delims;
+	int			error;
+	size_t		count;
+} t_delims;
+
+typedef struct s_string
+{
+	char	*str;
+} t_string;
+
+typedef struct s_string_array
+{
+	t_string	*strs;
+	int			error;
+	size_t		count;
+} t_strings;
+
+
+char	**ft_split_buf(const char *s, const void *delim_arr, int *sz);
```

Comment:
Why double declaration? Having it only in strings.h is enough. This way, if you decide to change anything later, you;d have to change this in two places.

Reply:
You are right